plasma/plasma-mvp/specifications/morevp.md

```diff
@@ -0,0 +1,655 @@
+Therefore, every exit, no matter the protocol used, must be included in the same priority queue for processing.
+
+
+## Alice-Bob Scenarios
+
+### Alice & Bob are honest and cooperating:
+
+1. Alice spends `UTXO1` in `TX1` to Bob, creating `UTXO2`.
+2. `TX1` is in-flight.
+3. Operator begins withholding blocks while `TX1` is still in-flight.
+Neither Alice nor Bob know if the transaction has been included in a block.
+4. Someone with access to `TX1` (Alice, Bob, or otherwise) starts an exit referencing `TX1` and places `exit bond`.
+5. Alice and Bob piggyback onto the exit and each place `piggyback bond`.
+6. Alice is honest, so she hasn’t spent `UTXO1` in any transaction other than `TX1`.
+7. After period 2, Bob receives the value of `UTXO2`, Alice does not receive the value of `UTXO1`.
+All bonds are refunded.
+
+
+### Mallory tries to exit a spent output:
+
+1. Alice spends `UTXO1` in `TX1` to Mallory, creating `UTXO2`.
+2. `TX1` is included in block `N`.
+3. Mallory spends `UTXO2` in `TX2`.
+4. Mallory starts an exit referencing `TX1` and places `exit bond`.
+5. Mallory piggybacks onto the exit and places `piggyback bond`.
+6. In period 2, someone reveals `TX2` spending `UTXO2`.
+This challenger receives Mallory’s `piggyback bond`.
+7. Alice is honest, so she hasn’t spent `UTXO1` in any transaction other than `TX1`.
+8. After period 2, Mallory’s `exit bond` is refunded, no one exits any UTXOs.
+
+
+### Mallory double spends her input:
+
+1. Mallory spends `UTXO1` in `TX1` to Bob, creating `UTXO2`.
+2. `TX1` is in-flight.
+3. Operator begins withholding blocks while `TX1` is still in-flight.
+Neither Mallory nor Bob know if the transaction has been included in a block.
+4. Mallory spends `UTXO1` in `TX2`.
+5. `TX2` is included in a withheld block.
+`TX1` is not included in a block.
+6. Bob starts an exit referencing `TX1` and places `exit bond`.
+7. Bob piggybacks onto the exit and places `piggyback bond`.
+8. In period 1, someone challenges the canonicity of `TX1` by revealing `TX2`.
+9. No one is able to respond to the challenge in period 2, so `TX1` is determined to be non-canonical.
+10. After period 2, Bob’s `piggyback bond` is refunded, no one exits any UTXOs.
+The challenger receives Bob’s `exit bond`.
+
+
+### Mallory spends her input again later:
+
+1. Mallory spends `UTXO1` in `TX1` to Bob, creating `UTXO2`.
+2. `TX1` is included in block `N`.
+3. Mallory spends `UTXO1` in `TX2`.
+4. `TX2` is included in block `N+M`
+5. Mallory starts an exit referencing `TX1` and places `exit bond`.
+6. In period 1, someone challenges the canonicity of `TX1` by revealing `TX2`.
+7. In period 2, someone responds to the challenge by proving that `TX1` was included before `TX2`.
+8. After period 2, the user who responded to the challenge receives Mallory’s `exit bond`, no one exits any UTXOs.
+
+
+### Mallory attempts to exit a spent input:
+
```

Comment:
I think the better description of this is she attempts to exit inputs that haven't been included yet. If at least one of them was spent and included then she wouldn't be losing both exit bonds.